From 3cf7211e4ca1ef6cd163f698867985e8c58d591b Mon Sep 17 00:00:00 2001
From: David S. Wang <dsw@cloudera.com>
Date: Wed, 27 Mar 2013 15:50:02 -0700
Subject: [PATCH 35/96] HBASE-6622 TestUpgradeFromHFileV1ToEncoding#testUpgrade fails in trunk

Reason: Test failure
Author: Ted Yu
Ref: CDH-9833
(cherry picked from commit e52671bfbcfda7b55527a4dd3217e4b0383c3d93)
---
 .../org/apache/hadoop/hbase/TestMultiVersions.java |    6 ++++--
 .../encoding/TestUpgradeFromHFileV1ToEncoding.java |    6 ++++--
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/src/test/java/org/apache/hadoop/hbase/TestMultiVersions.java b/src/test/java/org/apache/hadoop/hbase/TestMultiVersions.java
index cf2b5cd..19d6846 100644
--- a/src/test/java/org/apache/hadoop/hbase/TestMultiVersions.java
+++ b/src/test/java/org/apache/hadoop/hbase/TestMultiVersions.java
@@ -58,9 +58,11 @@ public class TestMultiVersions {
   private static final HBaseTestingUtility UTIL = new HBaseTestingUtility();
   private HBaseAdmin admin;
 
+  private static final int NUM_SLAVES = 3;
+
   @BeforeClass
   public static void setUpBeforeClass() throws Exception {
-    UTIL.startMiniCluster();
+    UTIL.startMiniCluster(NUM_SLAVES);
   }
 
   @AfterClass
@@ -140,7 +142,7 @@ public class TestMultiVersions {
     table.close();
     UTIL.shutdownMiniHBaseCluster();
     LOG.debug("HBase cluster shut down -- restarting");
-    UTIL.startMiniHBaseCluster(1, 1);
+    UTIL.startMiniHBaseCluster(1, NUM_SLAVES);
     // Make a new connection.  Use new Configuration instance because old one
     // is tied to an HConnection that has since gone stale.
     table = new HTable(new Configuration(UTIL.getConfiguration()), tableName);
diff --git a/src/test/java/org/apache/hadoop/hbase/io/encoding/TestUpgradeFromHFileV1ToEncoding.java b/src/test/java/org/apache/hadoop/hbase/io/encoding/TestUpgradeFromHFileV1ToEncoding.java
index b525409..93ea84e 100644
--- a/src/test/java/org/apache/hadoop/hbase/io/encoding/TestUpgradeFromHFileV1ToEncoding.java
+++ b/src/test/java/org/apache/hadoop/hbase/io/encoding/TestUpgradeFromHFileV1ToEncoding.java
@@ -52,12 +52,14 @@ public class TestUpgradeFromHFileV1ToEncoding {
   private static final int NUM_HFILE_V1_BATCHES = 10;
   private static final int NUM_HFILE_V2_BATCHES = 20;
 
+  private static final int NUM_SLAVES = 3;
+
   @BeforeClass
   public static void setUpBeforeClass() throws Exception {
     // Use a small flush size to create more HFiles.
     conf.setInt(HConstants.HREGION_MEMSTORE_FLUSH_SIZE, 1024 * 1024);
     conf.setInt(HFile.FORMAT_VERSION_KEY, 1); // Use HFile v1 initially
-    TEST_UTIL.startMiniCluster();
+    TEST_UTIL.startMiniCluster(NUM_SLAVES);
     LOG.debug("Started an HFile v1 cluster");
   }
 
@@ -81,7 +83,7 @@ public class TestUpgradeFromHFileV1ToEncoding {
     TEST_UTIL.shutdownMiniHBaseCluster();
 
     conf.setInt(HFile.FORMAT_VERSION_KEY, 2);
-    TEST_UTIL.startMiniHBaseCluster(1, 1);
+    TEST_UTIL.startMiniHBaseCluster(1, NUM_SLAVES);
     LOG.debug("Started an HFile v2 cluster");
     admin = new HBaseAdmin(conf);
     htd = admin.getTableDescriptor(TABLE_BYTES);
-- 
1.7.0.4

